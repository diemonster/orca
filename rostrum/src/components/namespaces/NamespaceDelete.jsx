import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { namespaceDelete, namespaceDeleteChangeInput } from '../../actions/namespaces';

class NamespaceDelete extends React.Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    event.preventDefault();

    const { dispatchNamespaceDeleteChangeInput } = this.props;

    dispatchNamespaceDeleteChangeInput(event.target.value);
  }

  handleSubmit(event) {
    event.preventDefault();

    const { dispatchNamespaceDelete, namespaceDeleteInput } = this.props;

    dispatchNamespaceDelete(namespaceDeleteInput);
  }

  render() {
    const { namespaceDeleteInput } = this.props;

    return (
      <div className="create-namespace">
        <h2>Delete Namespace</h2>
        <form className="namespace-input-form" onSubmit={this.handleSubmit}>
          <label className="namespace-label" htmlFor="namespace">
            Delete Namespace:
            <input
              type="text"
              value={namespaceDeleteInput}
              onChange={this.handleChange}
              placeholder="enter namespace"
              id="namespace"
            />
            <button
              className="button"
              type="submit"
              value="submit"
            >
              Delete
            </button>
          </label>
        </form>
      </div>
    );
  }
}

NamespaceDelete.propTypes = {
  dispatchNamespaceDelete: PropTypes.func.isRequired,
  dispatchNamespaceDeleteChangeInput: PropTypes.func.isRequired,
  namespaceDeleteInput: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  namespaceDeleteInput: state.namespace.namespaceDeleteInput,
});

const mapDispatchToProps = dispatch => ({
  dispatchNamespaceDelete: name => dispatch(namespaceDelete(name)),
  dispatchNamespaceDeleteChangeInput: namespaceDeleteInput => (
    dispatch(namespaceDeleteChangeInput(namespaceDeleteInput))
  ),
});

export default connect(mapStateToProps, mapDispatchToProps)(NamespaceDelete);
