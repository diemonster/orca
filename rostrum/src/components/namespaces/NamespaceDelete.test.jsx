import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


import { NamespaceDelete } from './NamespaceDelete';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  describe('NamespaceDelete', () => {
    it('should render itself and its subcomponents', () => {
      const props = {
        dispatchNamespaceDelete: jest.fn(),
        dispatchNamespaceDeleteChangeInput: jest.fn(),
        namespaceDeleteInput: '',
      };

      const namespaceDelete = shallow(<NamespaceDelete {...props} />);

      expect(namespaceDelete).toMatchSnapshot();
      expect(namespaceDelete.find('div').hasClass('delete-namespace')).toBe(true);
      expect(namespaceDelete.find('h2').text()).toEqual('Delete Namespace');
      expect(namespaceDelete.find('form').hasClass('namespace-input-form')).toBe(true);
      expect(namespaceDelete.find('form').prop('id')).toEqual('namespace-delete-input-form');
      expect(namespaceDelete.find('label').hasClass('namespace-input-label')).toBe(true);
      expect(namespaceDelete.find('label').text()).toEqual('Delete Namespace:');
      expect(namespaceDelete.find('input').prop('type')).toEqual('text');
      expect(namespaceDelete.find('input').prop('value')).toEqual('');
      expect(namespaceDelete.find('input').prop('placeholder')).toEqual('enter namespace');
      expect(namespaceDelete.find('input').prop('name')).toEqual('namespace-delete-input');
      expect(namespaceDelete.find('input').prop('id')).toEqual('namespace-delete-input');
      expect(namespaceDelete.find('button').hasClass('button')).toBe(true);
      expect(namespaceDelete.find('button').prop('type')).toEqual('submit');
      expect(namespaceDelete.find('button').prop('value')).toEqual('submit');
      expect(namespaceDelete.find('button').text()).toEqual('Delete');
    });

    it('appropriately calls dispatchNamespaceDeleteChangeInput() on input change', () => {
      const mockFn = jest.fn();
      const props = {
        dispatchNamespaceDelete: jest.fn(),
        dispatchNamespaceDeleteChangeInput: mockFn,
        namespaceDeleteInput: '',
      };

      const namespaceDelete = shallow(<NamespaceDelete {...props} />);

      const event = { preventDefault: jest.fn(), target: { value: 'a' } };
      namespaceDelete.find('input').simulate('change', event);
      expect(mockFn).toBeCalledWith('a');
    });

    it('appropriately calls dispatchNamespaceDelete() on form submit', () => {
      const mockFn = jest.fn();
      const props = {
        dispatchNamespaceDelete: mockFn,
        dispatchNamespaceDeleteChangeInput: jest.fn(),
        namespaceDeleteInput: 'some-namespace',
      };

      const namespaceDelete = shallow(<NamespaceDelete {...props} />);

      const event = { preventDefault: jest.fn(), target: { value: 'some-namespace' } };
      namespaceDelete.find('form').simulate('submit', event);
      expect(mockFn).toBeCalledWith('some-namespace');
    });
  });
});
