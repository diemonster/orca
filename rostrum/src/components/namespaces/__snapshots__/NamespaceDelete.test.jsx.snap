// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components NamespaceDelete should render itself and its subcomponents 1`] = `
<div
  className="delete-namespace"
>
  <h2>
    Delete Namespace
  </h2>
  <form
    className="namespace-input-form"
    id="namespace-delete-input-form"
    onSubmit={[Function]}
  >
    <label
      className="namespace-input-label"
      htmlFor="namespace-delete-input"
    >
      Delete Namespace:
    </label>
    <input
      id="namespace-delete-input"
      name="namespace-delete-input"
      onChange={[Function]}
      placeholder="enter namespace"
      type="text"
      value=""
    />
    <button
      className="button"
      type="submit"
      value="submit"
    >
      Delete
    </button>
  </form>
</div>
`;
